app/models/user.rb

```diff
@@ -34,17 +33,87 @@ class User < ApplicationRecord
          password_length: 10..128,
          otp_secret_encryption_key: ENV['ENCRYPTION_KEY'],
          otp_number_of_backup_codes: 10
-  # has_secure_password # not needed with devise
+
+  devise :omniauthable, omniauth_providers: [:okta] if ENV['OKTA_DOMAIN'].present?
+
+  def self.from_omniauth(auth)
+    user = find_by(
+      provider: auth['provider'],
```

Comment:
I'm tempted to push all of these changes into a concern for external_idp?  Does that make sense?

Reply:
Yeah.. It thought about it too and did i think in the cognito variant. Will do